core/src/main/java/com/amazon/opendistroforelasticsearch/sql/expression/datetime/DateTimeFormatterUtil.java

```diff
@@ -0,0 +1,131 @@
+      .put("%U", (date) -> // %U Week where Sunday is the first day - WEEK() mode 0
+          String.format("'%d'", CalendarLookup.getWeekNumber(0, date.toLocalDate())))
+      .put("%u", (date) -> // %u Week where Monday is the first day - WEEK() mode 1
+          String.format("'%d'", CalendarLookup.getWeekNumber(1, date.toLocalDate())))
+      .put("%V", (date) -> // %V Week where Sunday is the first day - WEEK() mode 2 used with %X
+          String.format("'%d'", CalendarLookup.getWeekNumber(2, date.toLocalDate())))
+      .put("%v", (date) -> // %v Week where Monday is the first day - WEEK() mode 3 used with %x
+          String.format("'%d'", CalendarLookup.getWeekNumber(3, date.toLocalDate())))
+      .put("%X", (date) -> // %X Year for week where Sunday is the first day, 4 digits used with %V
+          String.format("'%d'", CalendarLookup.getYearNumber(2, date.toLocalDate())))
+      .put("%x", (date) -> // %x Year for week where Monday is the first day, 4 digits used with %v
+          String.format("'%d'", CalendarLookup.getYearNumber(3, date.toLocalDate())))
+      .build();
+
+  private static final String MOD = "%";
+  private static final String QUOTE_LITERAL = "'";
+
+  private DateTimeFormatterUtil() {
+  }
+
+  /**
+   * Format the date using the date format String.
+   * @param dateExpr the date ExprValue of Date/Datetime/Timestamp/String type.
+   * @param formatExpr the format ExprValue of String type.
+   * @return Date formatted using format and returned as a String.
+   */
+  static ExprValue getFormattedDate(ExprValue dateExpr, ExprValue formatExpr) {
+    String format = formatExpr.stringValue();
+    final LocalDateTime date = dateExpr.datetimeValue();
+    for (Map.Entry<String, DateTimeFormatHandler> handler: HANDLERS.entrySet()) {
+      if (format.contains(handler.getKey())) {
+        format = format.replace(handler.getKey(), handler.getValue().getFormat(date));
+      }
+    }
+    format = literalReplace(format);
+
+    // English Locale matches SQL requirements.
+    // 'AM'/'PM' instead of 'a.m.'/'p.m.'
+    // 'Sat' instead of 'Sat.' etc
+    return new ExprStringValue(date.format(DateTimeFormatter.ofPattern(format, Locale.ENGLISH)));
+  }
+
+  /**
+   * Returns English suffix of incoming value.
+   * @param val Incoming value.
+   * @return English suffix as String (st, nd, rd, th)
+   */
+  private static String getSuffix(int val) {
+    // The numbers 11, 12, and 13 do not follow general suffix rules.
+    if ((SUFFIX_SPECIAL_START_TH <= val) && (val <= SUFFIX_SPECIAL_END_TH)) {
+      return SUFFIX_SPECIAL_TH;
+    }
+    return SUFFIX_CONVERTER.getOrDefault(val % 10, SUFFIX_SPECIAL_TH);
+  }
+
+  /**
+   * Goes through format String and replaces any %x with 'x' where x is unmapped.
+   * @param format Incoming format String without mapping completed.
+   * @return Outgoing format String with mapping completed.
+   */
+  private static String literalReplace(String format) {
```

Comment:
What is the expected result if the user input the format string which we doesn't support?

Reply:
Do you mean a format we don't support, such as %z? 
According to the documentation for MySQL it should come out as just z, so using the DateTimeFormatter specification, to get a literal out, we wrap in ', so %z => 'z'.

Reply:
Got it, does it covered in the test cases also?

Reply:
Yes